virttest/env_process.py

```diff
@@ -62,6 +62,14 @@
 _setup_manager = test_setup.SetupManager()
 
 
+#: Hooks to use for own customization stages of the virtual machines
+#: with params and env as supplied arguments
+preprocess_vm_off_hook = None
```

Comment:
the hooks are initialized here, but how to customize or assign this hooks with ?

Reply:
Here is a real life example from our plugin:

```python
# add env processing hooks to handle online/offline state get/set operations
env_process.preprocess_vm_on_hook = state_setup.get_state
def set_online_state(fn):
    def wrapper(run_params, env):
        run_params["skip_types"] = "offline"
        fn(run_params, env)
    return wrapper
def set_offline_state(fn):
    def wrapper(run_params, env):
        run_params["skip_types"] = "online ramdisk"
        fn(run_params, env)
    return wrapper
env_process.postprocess_vm_on_hook = set_online_state(state_setup.set_state)
env_process.postprocess_vm_off_hook = set_offline_state(state_setup.set_state)
```

I don't think this should really be described as hooks are meant to be replaced by customized functions. Or am I missing something?